lib/ui/window/platform_message_response_dart.cc

```diff
@@ -17,6 +19,10 @@ static std::atomic<uint64_t> platform_message_counter = 1;
 
 namespace flutter {
 namespace {
+void FreeFinalizer(void* isolate_callback_data, void* peer) {
+  free(isolate_callback_data);
```

Comment:
I don't think you want to free `isolate_callback_data` here. This data is basically some state that the embedder associates with an isolate(group) and is passed here for context (in case the finalizer needs access to isolate(group) related information).

The actual thing to be freed should always be `peer` (which is set below as `response.value.as_external_typed_data.peer`)